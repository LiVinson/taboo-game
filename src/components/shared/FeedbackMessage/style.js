import styled from 'styled-components'

export const StyledErrorMessage = styled.p`
	font-size: ${(props) => (props.large ? '2.2rem' : '1.5rem')};
	font-style: ${(props) => (props.large ? 'inherit' : 'italic')};
	color: ${(props) => (props.large || props.light ? props.theme.color.grayDark1 : 'red')};
	background-color: ${(props) => props.light && props.theme.color.offWhite};
`

export const StyledSuccessMessage = styled(StyledErrorMessage)`
	color: green;
`
