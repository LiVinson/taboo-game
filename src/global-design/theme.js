const theme = {
  color: {
    primary: "#1982C4", //light blue
    accent1: "#8AC926", //lime green
    accent2: "#ee5b2c", //orange
    accent3: "#a23b72", //purple
    primaryLight: "#58b1e9", //light-blue
    white: "#fff",
    offWhite: "#f7f9f9",
    lightGray1: "#f1f1f1",
    lightGray2: "#ebebeb",
    grayDark1: "#757575",
    grayDark2: "#333333"
  },
  text: {
    display: "'Oswald', sans-serif",
  },
  fontSize: {
    xsmall: "1.8rem",
    small: "2.2rem",
    medium: "2.7rem",
    large: "4rem",
    xlarge: "7rem",
  },
}


export default theme
