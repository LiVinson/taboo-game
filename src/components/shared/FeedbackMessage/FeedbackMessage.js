import React from "react"
import PropTypes from "prop-types"
import {StyledErrorMessage, StyledSuccessMessage} from "./style"

export const ErrorMessage = ({error, large, light}) => {
    return <StyledErrorMessage light={light} large={large}>{error}</StyledErrorMessage>
}

ErrorMessage.propTypes = {
    error: PropTypes.string,
    large: PropTypes.bool,
    light: PropTypes.bool
}

ErrorMessage.defaultProp = {
    large: false,
    error: "A problem has occured. Please try again."
}

export const SuccessMessage = ({message}) => {
    return <StyledSuccessMessage>{message}</StyledSuccessMessage>
}



SuccessMessage.propTypes = {
    message: PropTypes.string.isRequired,
}
