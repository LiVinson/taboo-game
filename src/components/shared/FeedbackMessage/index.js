import { ErrorMessage, SuccessMessage } from "./FeedbackMessage"

export {ErrorMessage, SuccessMessage}