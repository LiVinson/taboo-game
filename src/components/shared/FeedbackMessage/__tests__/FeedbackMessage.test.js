import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { ThemeProvider } from 'styled-components'
import theme from '../../../../global-design/theme'
import { SuccessMessage, ErrorMessage } from '../FeedbackMessage'

describe('ErrorMessage', () => {
	const defaultProps = {
		error: 'Something has gome wrong.',
	}

	test('Renders correctly when large and light props not provided', () => {
		const props = {
			...defaultProps,
		}
		const wrapper = renderer
			.create(
				<ThemeProvider theme={theme}>
					<ErrorMessage {...props} />
				</ThemeProvider>
			)
			.toJSON()
		expect(wrapper).toMatchSnapshot()
	})

	test('Renders correctly when large and light props  provided', () => {
		const props = {
			...defaultProps,
			large: true,
			light: true,
		}
		const wrapper = renderer
			.create(
				<ThemeProvider theme={theme}>
					<ErrorMessage {...props} />
				</ThemeProvider>
			)
			.toJSON()
		expect(wrapper).toMatchSnapshot()
	})
})

describe('SuccessMessage', () => {
	const defaultProps = {
		message: 'Scores updated!',
	}

	test('Renders correctly', () => {
		const props = {
			...defaultProps,
		}
		const wrapper = renderer
			.create(
				<ThemeProvider theme={theme}>
					<SuccessMessage {...props} />
				</ThemeProvider>
			)
			.toJSON()
		expect(wrapper).toMatchSnapshot()
	})
})
