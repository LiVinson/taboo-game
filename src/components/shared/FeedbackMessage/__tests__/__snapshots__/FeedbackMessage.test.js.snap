// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage Renders correctly when large and light props  provided 1`] = `
.c0 {
  font-size: 2.2rem;
  font-style: inherit;
  color: #757575;
  background-color: #f7f9f9;
}

<p
  className="c0"
>
  Something has gome wrong.
</p>
`;

exports[`ErrorMessage Renders correctly when large and light props not provided 1`] = `
.c0 {
  font-size: 1.5rem;
  font-style: italic;
  color: red;
}

<p
  className="c0"
>
  Something has gome wrong.
</p>
`;

exports[`SuccessMessage Renders correctly 1`] = `
.c0 {
  font-size: 1.5rem;
  font-style: italic;
  color: red;
  color: green;
}

<p
  className="sc-AxjAm c0"
>
  Scores updated!
</p>
`;
